homeassistant/components/bsblan/climate.py

```diff
@@ -40,130 +41,119 @@
     PRESET_NONE,
 ]
 
-HA_STATE_TO_BSBLAN = {
-    HVACMode.AUTO: "1",
-    HVACMode.HEAT: "3",
-    HVACMode.OFF: "0",
-}
-
-BSBLAN_TO_HA_STATE = {value: key for key, value in HA_STATE_TO_BSBLAN.items()}
-
-HA_PRESET_TO_BSBLAN = {
-    PRESET_ECO: "2",
-}
-
-BSBLAN_TO_HA_PRESET = {
-    2: PRESET_ECO,
-}
-
 
 async def async_setup_entry(
     hass: HomeAssistant,
     entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up BSBLan device based on a config entry."""
-    bsblan: BSBLan = hass.data[DOMAIN][entry.entry_id][DATA_BSBLAN_CLIENT]
-    info = await bsblan.info()
-    async_add_entities([BSBLanClimate(entry.entry_id, bsblan, info)], True)
+    """Set up BSBLAN device based on a config entry."""
+    data: HomeAssistantBSBLANData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            BSBLANClimate(
+                data.coordinator,
+                data.client,
+                data.device,
+                data.info,
+                entry,
+            )
+        ],
+        True,
+    )
 
 
-class BSBLanClimate(ClimateEntity):
-    """Defines a BSBLan climate device."""
+class BSBLANClimate(BSBLANEntity, CoordinatorEntity, ClimateEntity):
+    """Defines a BSBLAN climate device."""
 
-    _attr_supported_features = (
-        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
-    )
-    _attr_hvac_modes = HVAC_MODES
-    _attr_preset_modes = PRESET_MODES
+    coordinator: DataUpdateCoordinator[State]
 
     def __init__(
         self,
-        entry_id: str,
-        bsblan: BSBLan,
+        coordinator: DataUpdateCoordinator,
+        client: BSBLAN,
+        device: Device,
         info: Info,
+        entry: ConfigEntry,
     ) -> None:
-        """Initialize BSBLan climate device."""
-        self._attr_available = True
-        self._store_hvac_mode = None
-        self.bsblan = bsblan
-        self._attr_name = self._attr_unique_id = info.device_identification
-        self._attr_device_info = DeviceInfo(
-            identifiers={(DOMAIN, info.device_identification)},
-            manufacturer="BSBLan",
-            model=info.controller_variant,
-            name="BSBLan Device",
+        """Initialize BSBLAN climate device."""
+        super().__init__(client, device, info, entry)
+        CoordinatorEntity.__init__(self, coordinator)
+        self._attr_unique_id = f"{format_mac(device.MAC)}-climate"
+        self._attr_name = device.name
```

Comment:
This is the "main" entity of the device, isn't it?
If yes, you don't need to give the entity its own name https://developers.home-assistant.io/blog/2022/07/10/entity_naming, https://developers.home-assistant.io/docs/core/entity/#entity-naming:
```suggestion
        self._attr_has_entity_name = True
```

In addition, make it a class attribute

Reply:
This can be set as an class attribute (instead of instance attribute) as well 😄